packages/examples/cvat/exchange-oracle/src/handlers/completed_escrows.py

```diff
@@ -30,382 +27,259 @@
-        *,
-        max_attempts: int | None = None,
-    ) -> io.RawIOBase:
-        """
-        Sometimes CVAT downloading can fail with the 500 error.
-        This function tries to repeat the export in such cases.
-        """
-
-        if max_attempts is None:
-            max_attempts = CronConfig.track_completed_escrows_max_downloading_retries
-
-        attempt = 0
-        while attempt < max_attempts:
-            try:
-                return download_callback()
-            except cvat_api.exceptions.ApiException as e:
-                if 500 <= e.status < 600 and attempt + 1 < max_attempts:
-                    attempt += 1
-                    self.logger.info(
-                        f"Retrying downloading, attempt #{attempt} of {max_attempts}..."
-                    )
-                    retry_callback()
-                else:
-                    raise
-        return None
-
-    def _process_plain_escrows(self):
-        logger = self.logger
-
-        plain_task_types = [t for t in TaskTypes if t != TaskTypes.image_skeletons_from_boxes]
-        with SessionLocal.begin() as session:
-            completed_projects = cvat_service.get_projects_by_status(
-                session,
-                ProjectStatuses.completed,
-                included_types=plain_task_types,
-                limit=CronConfig.track_completed_escrows_chunk_size,
-                for_update=ForUpdateParams(skip_locked=True),
+    if max_attempts is None:
+        max_attempts = CronConfig.track_completed_escrows_max_downloading_retries
+
+    attempt = 0
+    while attempt < max_attempts:
+        try:
+            return download_callback()
+        except cvat_api.exceptions.ApiException as e:
+            if 500 <= e.status < 600 and attempt + 1 < max_attempts:
+                attempt += 1
+                logger.info(f"Retrying downloading, attempt #{attempt} of {max_attempts}...")
+                retry_callback()
+            else:
+                raise
+    return None
+
+
+def _export_escrow_annotations(
+    logger: logging.Logger,
+    chain_id: int,
+    escrow_address: str,
+    escrow_projects: Sequence[Project],
+    session: Session,
+) -> None:
+    manifest = parse_manifest(get_escrow_manifest(chain_id, escrow_address))
+
+    logger.debug(f"Downloading results for the escrow ({escrow_address=})")
+
+    jobs = cvat_service.get_jobs_by_escrow_address(session, escrow_address, chain_id)
+
+    annotation_format = CVAT_EXPORT_FORMAT_MAPPING[manifest.annotation.type]
+    job_annotations = _download_job_annotations(logger, annotation_format, jobs)
+
+    if manifest.annotation.type == TaskTypes.image_skeletons_from_boxes.value:
+        # we'll have to merge annotations ourselves for skeletons
+        # might want to make this the only behaviour in the future
+        project_annotations_file = None
+        project_images = None
+    else:
+        # escrows with simple task types are ought to have only one project
```

Comment:
```suggestion
        # escrows with simple task types must have only one project
```
I'd say it should be an assertion error instead. It's not like it's something ideal that never happens, which is supposed by "ought to".